src/internal/storedPixelDataToCanvasImageData.js

```diff
@@ -21,28 +21,51 @@
      */
     function storedPixelDataToCanvasImageData(image, lut, canvasImageDataData)
     {
+        var start = (window.performance ? performance.now() : Data.now());
         var pixelData = image.getPixelData();
+        image.stats.lastGetPixelDataTime = (window.performance ? performance.now() : Data.now()) - start;
+
         var minPixelValue = image.minPixelValue;
         var canvasImageDataIndex = 3;
         var storedPixelDataIndex = 0;
-        var localNumPixels = pixelData.length;
-        var localPixelData = pixelData;
-        var localLut = lut;
-        var localCanvasImageDataData = canvasImageDataData;
+        var numPixels = pixelData.length;
 
+        start = (window.performance ? performance.now() : Data.now());
         // NOTE: As of Nov 2014, most javascript engines have lower performance when indexing negative indexes.
         // We have a special code path for this case that improves performance.  Thanks to @jpambrun for this enhancement
-        if(minPixelValue < 0){
-            while(storedPixelDataIndex < localNumPixels) {
-                localCanvasImageDataData[canvasImageDataIndex] = localLut[localPixelData[storedPixelDataIndex++] + (-minPixelValue)]; // alpha
+
+        // Added two paths (Int16Array, Uint16Array) to avoid polymorphic deoptimization in chrome.
+        if (pixelData instanceof Int16Array) {
```

Comment:
I have added this fix as well, it prevents a significant slowdown when this function becomes polymorphic after being called on a Uint16 followed by a Int16 image (or vice versa). But it looks pretty stupid..